app/code/Ometria/Core/Helper/Cookiechannel.php

```diff
@@ -1,97 +1,185 @@
             $commands = array_filter($commands);
         }
 
         $commands[] = $str;
-        if (count($commands)>6) $commands = array_slice($commands, 0, 6);
+        if (count($commands) > 6) {
+            $commands = array_slice($commands, 0, 6);
+        }
 
         $commands = array_unique($commands);
         $commands = array_filter($commands);
         $commands = array_values($commands);
         sort($commands);
         $commands = array_values($commands);
 
-        $new_cookie = implode(self::SEPERATOR_BETWEEN_COMMANDS, $commands);
-        if (strlen($new_cookie)>1000) $new_cookie = '';
+        $newCookie = implode(self::SEPERATOR_BETWEEN_COMMANDS, $commands);
+        if (strlen($newCookie) > 1000) {
+            $newCookie = '';
+        }
 
-        if (!headers_sent() && ($new_cookie!=$existing_cookie)) {
-            $this->cookie_did_change = true;
-            $_COOKIE[self::COOKIE_NAME] = $new_cookie;
-            //setcookie(self::COOKIE_NAME, $new_cookie, 0, '/');
-            //if ($set_cookie) Mage::getModel('core/cookie')
-            //                    ->set(self::COOKIE_NAME, $new_cookie, 0, '/');
-            if ($set_cookie) $this->sendCookie();
+        if (!headers_sent() && ($newCookie != $existingCookie)) {
+            $this->cookieDidChange = true;
+            $_COOKIE[self::COOKIE_NAME] = $newCookie;
+
+            $this->sendCookie();
         }
     }
-    
-    public function sendCookie(){
-        if (!$this->cookie_did_change) return;
+
+    public function sendCookie()
+    {
+        if (!$this->cookieDidChange) {
+            return;
+        }
+
+        if ($this->helperConfig->isCookiebotEnabled() && $this->cookiebotCookieAllowed() === false) {
+            return;
+        }
+
         $cookie = isset($_COOKIE[self::COOKIE_NAME]) ? $_COOKIE[self::COOKIE_NAME] : '';
         setcookie(self::COOKIE_NAME, $cookie, 0, '/');
-        $this->cookie_did_change = false;
-    }    
-}
\ No newline at end of file
+        $this->cookieDidChange = false;
+    }
+
+    /**
+     * @return bool
+     */
+    private function cookiebotCookieAllowed()
+    {
+        $cookieAllowed = false;
+
+        if (isset($_COOKIE[self::COOKIEBOT_COOKIE_NAME])) {
+            switch ($_COOKIE[self::COOKIEBOT_COOKIE_NAME]) {
+                case "-1":
+                    //The user is not within a region that requires consent - all cookies are accepted
+                    $cookieAllowed = true;
+                    break;
+
+                default:
+                    // The user must give their consent
+                    $cookieConsent = $this->getCookiebotConsent();
+                    if ($cookieConsent) {
+                        $cookieClass = $this->helperConfig->getCookiebotClass();
+                        // Consent cookie was found
+                        if (isset($cookieConsent[$cookieClass]) && filter_var($cookieConsent[$cookieClass], FILTER_VALIDATE_BOOLEAN)) {
```

Comment:
Do we need to check if the key exists as a first check?

`if (array_key_exists($cookieClass, $cookieConsent) && ...`

Reply:
Actually, as you have wrapped the $cookieConsent in a truthy check should be ok